LoopCore/LoopSettings.swift

```diff
@@ -181,16 +174,15 @@ extension LoopSettings {
         preMealOverride = nil
     }
 
-    public func legacyWorkoutOverride(beginningAt date: Date = Date(), for duration: TimeInterval) -> TemporaryScheduleOverride? {
+    public mutating func legacyWorkoutOverride(beginningAt date: Date = Date(), for duration: TimeInterval) -> TemporaryScheduleOverride? {
```

Comment:
Instead of having to make this `mutating`, can you move the `indefiniteWorkoutOverrideEnabledDate` state into your new `LoopSettingsManager`?

Reply:
After sleep, I decided to follow your original proposal. Given that, the LoopSettingsAlerter would track the usage of the settings nor do I think it should handle storing/restoring. What do you think?